Add listing and playing recorded interactive sessions to tsh docs (#18792)
gravitational/teleport: docs/pages/reference/cli.mdx

$ tsh --user=joe --login=ec2-user join <session-id>
```

### tsh recordings ls

List recorded sessions.

```code
$ tsh recordings ls [<flags>]
```

#### Flags

| Name | Default Value(s) | Allowed Value(s) | Description |
| - | - | - | - |
| `--from-utc` | 24 hours ago | date | Start of time range in which recordings are listed. Format 2006-01-02. Defaults to 24 hours ago. |
| `--to-utc` | current | date | Start of time range in which recordings are listed. Format 2006-01-02. Defaults to 24 hours ago. |
| `--limit` | 50 | number | Maximum number of recordings to show. |
| `--last` | none | duration | Duration into the past from which session recordings should be listed. Format 5h30m40s |

> #### Global flags

Reviewer: Not a question for this PR, but I think it's silly that we duplicate this section for every single command.

cc @alexfornuto @ptgott

Reviewer 2: Agreed. I haven't been around long enough to know what convention or conversation (if any) led to this being the case, but I think it's worth revisiting.

These flags are available for all commands `--login, --proxy, --user, --ttl, --identity, --cert-format, --insecure, --auth, --skip-version-check, --debug, --jumphost, --format`.
Run `tsh help <subcommand>` or see the [Global Flags section](#tsh-global-flags).

#### Examples

```code
# get the recorded sessions from the last 24 hours
$ tsh --proxy proxy.example.com recordings ls
ID                                   Type Participants Hostname Timestamp
------------------------------------ ---- ------------ -------- -------------------
b0a04442-70dc-4be8-9308-7b7901d2d600 ssh  jeff         dev       Nov 26 16:36:16 UTC
c0a02222-70dc-4be8-9308-7b7901d2d600 kube alice                  Nov 26 20:36:16 UTC
d0a04442-70dc-4be8-9308-7b7901d2d600 ssh  navin        test      Nov 26 16:36:16 UTC

> #After retrieving the recorded session can be played with tsh play

Reviewer: ```suggestion
# The session can be played with tsh play
```

$ tsh play c0a02222-70dc-4be8-9308-7b7901d2d60

> # Retrieve recorded sessions that occurred between Nov 1, 2022 to Nov 3, 2022

Reviewer: ```suggestion
# List recorded sessions that occurred between Nov 1, 2022 to Nov 3, 2022
```

"retrieve" sounds like "download" to me. This is just a listing, it's not really "retrieving" anything

$ tsh recordings ls --from-utc=2022-11-01 --to-utc=2022-11-3

# Retrieve recorded sessions in the last 6 hours 
$ tsh recordings ls --last=6h0m0s
```

> <Admonition

Reviewer: This might make more sense in a troubleshooting guide if we have one.

Author: @alexfornuto thoughts?

Reviewer 2: We don't have a troubleshooting page for the Reference section. We could:
- Add this into a troubleshooting section for the existing [Audit Events and Records](https://goteleport.com/docs/reference/audit/) page,
- Incorporate it into Zac's [drafted](https://github.com/gravitational/teleport/pull/18717) overview of session recording,
- Just keep it here.

Author: @alexfornuto I would just leave it in here

Reviewer 2: Fine by me. Consider using a [type](http://localhost:3000/docs/contributing/documentation/reference/#admonitions) other than `note`.

$ tsh recordings ls --last=6h0m0s
```

<Admonition
  type="note"
  title="Recorded Sessions Availability"
  scope={["oss", "enterprise"]} scopeOnly
>

Recorded sessions are linked from the audit events to session recordings files. Self-hosted high availability Teleport

> deployments with directory storage for audit logs and session recordings can result in unavailable playback. That occurs

Reviewer 2: >Self-hosted high availability Teleport
deployments with directory storage for audit logs and session recordings can result in unavailable playback. 

Can we rephrase? This sentence is long and hard to parse. Not sure how to edit or I'd offer a suggestion.

Worth noting though that in my case the issue wasn't with an HA setup and separate storage locations. Instead I suspect the issue was with rotation and older sessions being archived/compressed.

Author: rephrased @alexfornuto

since each auth server has its own audit log and recorded session records.

```code
$ tsh play c8e1b2c5-322a-4095-89e3-391edfd2da9b
ERROR: Recording for session c8e1b2c5-322a-4095-89e3-391edfd2da9b not found.
```

> Using a SIEM that combines the audit logs will help consolidate the list of available recordings. Downloading the 

Reviewer 2: Is it safe to assume that all readers of this page will be familiar with the acronym SIEM?

Author: provided

> Using a SIEM that combines the audit logs will help consolidate the list of available recordings. Downloading the 
> recorded session is directly playable as a file. 

Reviewer 2: ```suggestion
Using a SIEM that combines the audit logs will help consolidate the list of available recordings. Downloaded recorded session are directly playable as a file. 
```

  scope={["oss", "enterprise"]} scopeOnly
>

> Recorded sessions are linked from the audit events to session recordings files in their [storage backend](../reference/backends.mdx).
> The following error can occur if a session recording file is not available or when employing multiple auth servers with directory storage
> backend for recorded sessions. When using directory storage backend for audit logs and recorded sessions then only the auth server with that
> recorded session can retrieve it.

Reviewer 2: Line breaks and minor copy edits.
```suggestion
Recorded sessions are linked from the audit events to session recordings files
in their [storage backend](../reference/backends.mdx).
The following error can occur if a session recording file is not available or
when employing multiple auth servers with directory storage backend for recorded
sessions. When using a directory storage backend for audit logs and recorded sessions,
only the auth server with that recorded session can retrieve it.
```

```code
$ tsh play c8e1b2c5-322a-4095-89e3-391edfd2da9b
ERROR: Recording for session c8e1b2c5-322a-4095-89e3-391edfd2da9b not found.
```

> Using a Security information and event management (SIEM) service that combines the audit logs will help consolidate the list of available recordings. Downloaded recorded session are directly playable as a file. 

Reviewer 2: - Capitalized the words in the acronym
- Put the last sentence in a new paragraph since it begins a new topic.

```suggestion
Using a Security Information and Event Management (SIEM) service that combines
the audit logs will help consolidate the list of available recordings.

Downloaded recorded session are directly playable as a file.
```
